code/modules/research/research.dm

```diff
@@ -50,98 +50,112 @@ research holder datum.
 	var/list/known_designs = list()			//List of available designs.
 
 /datum/research/New()		//Insert techs into possible_tech here. Known_tech automatically updated.
-	for(var/T in typesof(/datum/tech) - /datum/tech)
-		known_tech += new T(src)
-	for(var/D in typesof(/datum/design) - /datum/design)
-		possible_designs += new D(src)
+	for(var/tech_path in subtypesof(/datum/tech))
+		var/datum/tech/T = new tech_path(src)
```

Comment:
This thing is repeated like 4 times. is there a way to make it only happen once instead?